test/functional/posts.js

```diff
@@ -260,69 +280,73 @@ describe("PostsController", function() {
           done()
         })
     })
+
+    it("should not update another user's post", function(done) {
+      var newBody = "New body"
+      request
+          .post(app.config.host + '/v1/posts/' + context.post.id)
+          .send({ post: { body: newBody },
+            authToken: otherUserAuthToken,
+            '_method': 'put'
+          })
+          .end(function(err, res) {
+            err.status.should.eql(403)
+            res.body.err.should.eql("You can't update another user's post")
+
+            done()
+          })
+    })
   })
 
   describe('#show()', function() {
-    var post
-      , authToken
+    var context = {}
 
-    beforeEach(funcTestHelper.createUser('Luna', 'password', function(token) {
-      authToken = token
-    }))
-
-    beforeEach(function(done) {
-      var body = 'Post body'
-      request
-        .post(app.config.host + '/v1/posts')
-        .send({ post: { body: body }, authToken: authToken })
-        .end(function(err, res) {
-          post = res.body.posts
-
-          done()
-        })
-    })
+    beforeEach(funcTestHelper.createUserCtx(context, 'Luna', 'password'))
+    beforeEach(funcTestHelper.createPost(context, 'Post body'))
 
     it('should show a post', function(done) {
       request
-        .get(app.config.host + '/v1/posts/' + post.id)
-        .query({ authToken: authToken })
+        .get(app.config.host + '/v1/posts/' + context.post.id)
+        .query({ authToken: context.authToken })
         .end(function(err, res) {
           res.body.should.not.be.empty
           res.body.should.have.property('posts')
           res.body.posts.should.have.property('body')
-          res.body.posts.body.should.eql(post.body)
+          res.body.posts.body.should.eql(context.post.body)
 
           done()
         })
     })
+
+    it('should return 404 given an invalid post ID', function(done) {
+      request
+          .get(app.config.host + '/v1/posts/123_no_such_id')
+          .query({ authToken: context.authToken })
+          .end(function(err, res) {
+            err.status.should.eql(404)
+            res.body.err.should.eql("Can't find post")
+
+            done()
+          })
+    })
   })
 
   describe('#destroy()', function() {
     var username = 'Luna'
-    var post
-      , authToken
+    var context = {}
+    var otherUserAuthToken
 
-    beforeEach(funcTestHelper.createUser(username, 'password', function(token) {
-      authToken = token
+    beforeEach(funcTestHelper.createUserCtx(context, username, 'password'))
+    beforeEach(funcTestHelper.createPost(context, 'Post body'))
+    beforeEach(funcTestHelper.createUser('yole', 'pw', function(token) {
```

Comment:
I actually think that the code will be much shorter if you move it to the opposite direction: when all setup functions save everything they create in the context object automatically. Then you can extract the data from the context object like, for example: `context.authTokens['yole']` or `context.posts[1]` in case the test setup creates multiple posts. And it's still going to be very readable.